### DISCUSSION TITLE
Correct way to auto-logout the client, and detect this event?

### DISCUSSION THREAD

**Discussion by BenjaminWFox-Lumedic** [nextauthjs/next-auth#776] [Help]
**Your question**
- Is there a best 'NextAuth' way to auto-logout the client?
- Is there any way to detect that this has happened via event or callback in the client?

(I think this is sort of the opposite of this issue https://github.com/nextauthjs/next-auth/issues/381)

**What are you trying to do**
In the scenario where a client window is open and the user steps away for some time I need to ensure that the user is explicitly logged out & sent to a log-out page. (like on banks/financial sites, healthcare sites, etc...).

Ideally there would be specific messaging that this logout was due to inactivity, which is where having an event or callback would be helpful, but in testing I couldn't find any way to accomplish this.

Currently I have:
- `keepAlive` set to 10s longer than `maxAge` which seems to work to log out the client. [The documentation states](https://next-auth.js.org/getting-started/client#options) you shouldn't do this, but I'm assuming that's only because the intention of `keepAlive` is to keep the session alive and it wasn't intended for this opposite purpose. Is there any problem with this?
> The value for keepAlive should always be lower than the value of the session maxAge option.

**Feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [x] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [ ] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by cha2hyun** [comment]
@BenjaminWFox-Lumedic 
Any help for same issue?

**Comment by SawkaDev** [comment]
Did you find a way to do this?
